src/proto/uprotocol/uauthority.rs

```diff
@@ -14,13 +14,95 @@
 use crate::uprotocol::uri::uauthority::Number;
 use crate::uprotocol::UAuthority;
 
-pub use crate::uri::validator::ValidationError;
+pub use crate::proto::uprotocol::uuri::SerializationError;
+use crate::uri::validator::ValidationError;
 
 const REMOTE_IPV4_BYTES: usize = 4;
 const REMOTE_IPV6_BYTES: usize = 16;
 const REMOTE_ID_MINIMUM_BYTES: usize = 1;
 const REMOTE_ID_MAXIMUM_BYTES: usize = 255;
 
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum AddressType {
+    Local = 0,
+    IPv4 = 1,
+    IPv6 = 2,
+    ID = 3,
+}
+
+impl AddressType {
+    pub fn value(self) -> u8 {
+        self as u8
+    }
+}
+
+impl TryFrom<u8> for AddressType {
+    type Error = SerializationError;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(AddressType::Local),
+            1 => Ok(AddressType::IPv4),
+            2 => Ok(AddressType::IPv6),
+            3 => Ok(AddressType::ID),
+            _ => Err(SerializationError::new(format!(
+                "unknown address type ID [{}]",
+                value
+            ))),
+        }
+    }
+}
+
+impl TryFrom<i32> for AddressType {
+    type Error = SerializationError;
+
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        if let Ok(v) = u8::try_from(value) {
+            Self::try_from(v)
+        } else {
+            Err(SerializationError::new(format!(
+                "unknown address type ID [{}]",
+                value
+            )))
+        }
+    }
+}
+
+impl TryFrom<&UAuthority> for AddressType {
+    type Error = SerializationError;
+
+    fn try_from(authority: &UAuthority) -> Result<Self, Self::Error> {
+        let address_type: AddressType;
+
+        if authority.has_id() {
+            address_type = AddressType::ID;
+        } else if authority.has_ip() {
+            match authority.ip().len() {
+                4 => address_type = AddressType::IPv4,
```

Comment:
Could use the definitions up above in here:
```rust
const REMOTE_IPV4_BYTES: usize = 4;
const REMOTE_IPV6_BYTES: usize = 16;
```

Reply:
Great catch, thank you - done!